drivers/i2c/i2c_ll_stm32_v2.c

```diff
@@ -61,6 +62,70 @@ void stm32_i2c_event_isr(void *arg)
 	struct i2c_stm32_data *data = DEV_DATA((struct device *)arg);
 	I2C_TypeDef *i2c = cfg->i2c;
 
+#if defined(CONFIG_I2C_SLAVE)
+	if (data->slave_attached) {
+		if (LL_I2C_IsActiveFlag_ADDR(i2c)) {
+			u32_t dir;
+
+			LL_I2C_ClearFlag_ADDR(i2c);
+
+			dir = LL_I2C_GetTransferDirection(i2c);
+			if (dir == LL_I2C_DIRECTION_WRITE) {
+				data->slave_funcs->write_request(
+							data->slave_priv);
+				LL_I2C_EnableIT_RX(i2c);
+			} else {
+				u8_t val;
+				data->slave_funcs->read_request(
+							data->slave_priv, &val);
+				LL_I2C_TransmitData8(i2c, val);
+				LL_I2C_EnableIT_TX(i2c);
+				LL_I2C_EnableIT_NACK(i2c);
+			}
+
+			LL_I2C_EnableIT_STOP(i2c);
+			LL_I2C_EnableIT_ERR(i2c);
+		} else if (LL_I2C_IsActiveFlag_NACK(i2c) ||
+			   LL_I2C_IsActiveFlag_STOP(i2c)) {
+			if (LL_I2C_IsActiveFlag_NACK(i2c)) {
+				LL_I2C_ClearFlag_NACK(i2c);
+			}
+
+			if (LL_I2C_IsActiveFlag_STOP(i2c)) {
+				LL_I2C_ClearFlag_STOP(i2c);
+			}
+
+			data->slave_funcs->stop(data->slave_priv);
+			
+			LL_I2C_DisableIT_NACK(i2c);
+			LL_I2C_DisableIT_TX(i2c);
+			LL_I2C_DisableIT_RX(i2c);
+			LL_I2C_DisableIT_STOP(i2c);
+			LL_I2C_DisableIT_ERR(i2c);
+			
+			LL_I2C_AcknowledgeNextData(i2c, LL_I2C_ACK);
+
+			/* Flush remaining TX byte */
+			LL_I2C_Disable(i2c);
+			LL_I2C_Enable(i2c);
+		} else if (LL_I2C_IsActiveFlag_TXIS(i2c)) {
+			u8_t val;
+			
+			data->slave_funcs->read_done(data->slave_priv);
+			data->slave_funcs->read_request(data->slave_priv, &val);
+			LL_I2C_TransmitData8(i2c, val);
+		} else if (LL_I2C_IsActiveFlag_RXNE(i2c)) {
+			u8_t val = LL_I2C_ReceiveData8(i2c);
+			
+			if (data->slave_funcs->write_done(data->slave_priv,
+							  val))
+				LL_I2C_AcknowledgeNextData(i2c, LL_I2C_NACK);
+		}
+
+		return;
```

Comment:
Could you move the logic from line 67 to here, into a dedicated function? just to get a clearer isr handler.

Reply:
It was my plan, will do